test/hotspot/jtreg/runtime/NMT/MallocLimitTest.java

```diff
@@ -0,0 +1,251 @@
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test id=global_limit
+ * @summary Verify -XX:MallocLimit with a global limit
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib
+ * @run driver MallocLimitTest global_limit
+ */
+
+/*
+ * @test id=compiler_limit
+ * @summary Verify -XX:MallocLimit with a compiler-specific limit (for "mtCompiler" category)
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib
+ * @run driver MallocLimitTest compiler_limit
+ */
+
+/*
+ * @test id=test_multi_limit
+ * @summary Verify -XX:MallocLimit with multiple limits
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib
+ * @run driver MallocLimitTest multi_limit
+ */
+
+/*
+ * @test id=test_valid_settings
+ * @summary Verify -XX:MallocLimit rejects invalid settings
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib
+ * @run driver MallocLimitTest valid_settings
+ */
+
+/*
+ * @test id=test_invalid_settings
+ * @summary Verify -XX:MallocLimit rejects invalid settings
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib
+ * @run driver MallocLimitTest invalid_settings
+ */
+
+/*
+ * @test id=test_limit_without_nmt
+ * @summary Verify that the VM warns if -XX:MallocLimit is given but NMT is disabled
+ * @modules java.base/jdk.internal.misc
+ * @library /test/lib
+ * @run driver MallocLimitTest limit_without_nmt
+ */
+
+
+import jdk.test.lib.Asserts;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+public class MallocLimitTest {
+
+    private static ProcessBuilder processBuilderWithSetting(String... extra_settings) {
+        String[] vmargs = new String[] {
+            "-XX:+UnlockDiagnosticVMOptions", // MallocLimit is diagnostic
+            "-Xmx64m", "-XX:-CreateCoredumpOnCrash", "-Xlog:nmt",
+            "-XX:NativeMemoryTracking=summary"
+        };
+        String[] vmargs2 = new String[] { "-version" };
+        String[] both = Stream.concat(Stream.concat(Arrays.stream(vmargs), Arrays.stream(extra_settings)), Arrays.stream(vmargs2))
+                .toArray(String[]::new);
+        ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(both);
+        return pb;
+    }
+
+    private static void test_global_limit() throws IOException {
+        long small_memory_size = 1024*1024; // 1m
+        ProcessBuilder pb = processBuilderWithSetting("-XX:MallocLimit=" + small_memory_size);
+        OutputAnalyzer output = new OutputAnalyzer(pb.start());
+        output.reportDiagnosticSummary();
```

Comment:
Do you really need `reportDiagnosticSummary()`? I thought it would be printed if `OutputAnalyzer` barfs on error.